addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/generators/DataGenerator.java

```diff
@@ -202,11 +211,23 @@ public boolean isArray(String type) {
         return "array".equals(type);
     }
 
+    public boolean isArray(Schema<?> schema) {
```

Comment:
`static` (same for `isMap`).

Reply:
Hi @thc202 , Why those methods need to be static? Could you clarify, please?

Reply:
Intention/clarity (they should be private also), those methods don't access/need the state of the instance, they are just utility/helper methods.

Reply:
Agree on to change from public to private because it makes sense but as static and I don't see the reason. The is isArray is used on ArrayGeneratorc class and If we want to treat them as utility/helper methods, I believe they should be moved a new class. What do you think? Just feels weird to have some methods private and static and others as public.

Reply:
Well, they are already being used as utility/helper methods, being static represents better what they do (i.e. don't access the instance's state).
I don't see a problem with some being private and others public, if they aren't needed by other classes then private is better than public but I'm ok if they are moved to other class.